src/factory/mod.rs

```diff
@@ -325,20 +325,38 @@ impl Factory {
             },
             _ => unreachable!(),
         };
-        let mat = new_mat.unwrap_or_else(|| match hub.nodes[&template.node].sub_node {
+        let material = match hub.nodes[&template.node].sub_node {
             SubNode::Visual(ref mat, _) => mat.clone(),
             _ => unreachable!(),
-        });
-        Mesh { object: hub.spawn_visual(mat, gpu_data) }
+        };
+        Mesh { object: hub.spawn_visual(material, gpu_data) }
+    }
+
+    /// Create a `Mesh` sharing the geometry with another one but with a different material.
+    /// Rendering a sequence of meshes with the same geometry is faster.
+    pub fn mesh_instance_with_material<M: Into<Material>>(
```

Comment:
This replaces the previous `Option<Material>` idea. I wasn't sure about how to handle the type parameter `fn mesh_instance<M: Into<Material>>(&mut self, template: &Mesh, material: Option<M>)` when `material` is `None`.